content/2017.04.getting-started-with-elm.rst

```diff
@@ -0,0 +1,433 @@
+
+Now that you know how to render your page in HTML, it is time to start
+a program that know how to handle events.
+
+The way elm works about that is by having:
+
+- a ``Model``, an Elm record: a bit like a Javascript object with
+  properties, that keep the state of the app,
+- an update function that will handle all the app events
+- a view function that will return the virtualdom that matches the state.
+
+The events and their params are defined in a ``Msg`` type. It is a bit
+like a enum that would take parameters.
+
+In order to do that we can use the ``beginnerProgram`` from the html package:
+
+.. code-block:: elm
+
+    module Main exposing (..)
+    
+    import Html exposing (..)
+    
+    
+    type Msg
+        = Noop
+    
+    
+    type alias Model =
+        { name : String }
+    
+    
+    main =
+        beginnerProgram { model = { name = "Rémy" }, view = view, update = update }
+    
+    
+    update : Msg -> Model -> Model
+    update msg model =
+        model
+    
+    
+    view : Model -> Html Msg
+    view model =
+        text ("Hello " ++ model.name)
+
+We can now handle an event and change the name when we click on it.
+
+.. code-block:: elm
+
+    module Main exposing (..)
+
+    import Html exposing (..)
+    import Html.Attributes exposing (..)
+    import Html.Events exposing (..)
+
+
+    type Msg
+        = Switch
+
+
+    type alias Model =
+        { name : String }
+
+
+    main =
+        beginnerProgram { model = { name = "Rémy" }, view = view, update = update }
+
+
+    update : Msg -> Model -> Model
+    update msg model =
+        case msg of
+            Switch ->
+                { model | name = "Séverine" }
+
+
+    view : Model -> Html Msg
+    view model =
+        div []
+            [ text "Hello "
+            , a [ href "#", onClick Switch ] [ text model.name ]
+            ]
+
+You can refresh the page and try it.
+
+If we want to switch back to ``Rémy`` when we click on ``Séverine`` we can add a ``if``:
+
+.. code-block:: elm
+
+    module Main exposing (..)
+
+    import Html exposing (..)
+    import Html.Attributes exposing (..)
+    import Html.Events exposing (..)
+
+
+    type Msg
+        = Switch
+
+
+    type alias Model =
+        { name : String }
+
+
+    main =
+        beginnerProgram { model = { name = "Rémy" }, view = view, update = update }
+
+
+    update : Msg -> Model -> Model
+    update msg model =
+        case msg of
+            Switch ->
+                if model.name == "Rémy" then
+                    { model | name = "Séverine" }
+                else
+                    { model | name = "Rémy" }
+
+
+    view : Model -> Html Msg
+    view model =
+        div []
+            [ text "Hello "
+            , a [ href "#", onClick Switch ] [ text model.name ]
+            ]
+
+
+Conclusion
+==========
+
+That's about it. You now know more than you think about Elm. I hope
+you can try it in a small project and start to like it as much as we
+do.
```

Comment:
I like `You now know more than you think about Elm`. I'm not sure about `I hope you can try it`; on the one hand it sounds correct but very polite, but on the other hand I'd probably just say `I hope you try it`.